library/core/src/intrinsics.rs

```diff
@@ -341,6 +341,7 @@ extern "rust-intrinsic" {
     /// [`Ordering::Relaxed`] as the `order`. For example, [`AtomicBool::load`].
     #[rustc_nounwind]
     pub fn atomic_load_relaxed<T: Copy>(src: *const T) -> T;
+    /// Do NOT use this intrinsic; "unordered" operations do not exist in our memory model!
```

Comment:
IIRC they were added for experimentation. Maybe the warning could be a bit more informative?

For whom is that warning anyway? std/compiler devs or for nightly users?
I.e. are you trying to convey that they shouldn't be exposed through stable std APIs or that nightly users are moving on thin ice when they experiment with this? Because the latter could be reasonable if they're doing that to experiment and demonstrate performance differences.

Reply:
I have clarified: in terms of the AM, I think the best that we can do is that these intrinsics act like regular non-atomic reads and writes.